pkg/fqdn/cache.go

```diff
@@ -108,7 +111,10 @@ type DNSCache struct {
 	// expired in forward, should also be added/removed in reverse.
 	reverse map[string]nameEntries
 
-	// LastCleanup is the last time that the cleanup happens.
+	// LastCleanup is the latest time for which entries have been expired. It is
+	// used as "now" when doing lookups and advanced by calls to .GC
+	// When an entry is added with an expiration time before lastCleanup, it is
+	// set to that value.
```

Comment:
It has to do with how (in this PR) cleanupTime is now used. Expiration is done by iterating over a map of expiration-time -> entries. The time is in seconds so it is somewhat grouped. `cleanupTime` historically limited the number of iterations (the older code derived this in the `timediff` variable) by setting the start of the iteration, and the `now` parameter set the end. This used to work fine because lookups also elided expired data so the external view of the cache was consistent in real-time (i.e. if the cleanup controller had not already executed, a lookup would already not include expired entries).
Now that I explicitly rely on `cleanupTime` a lookup may return expired entries (as intended) but I was worried that it would advance too far and never go back, thus never expiring that entry. I think it was also a way to set the value on first insert, but that has other solutions.